### ISSUE TITLE
Fails if not Python3.7 or later

### ISSUE THREAD

**Issue by ashishbijlani** [ossillate-inc/packj#3]
Reported by @[oefrha](https://news.ycombinator.com/user?id=oefrha) on HN:

```
- esprima==4.0.0 requires Python 3.6 (EOL) or lower because the package is really old and uses the async keyword (promoted in py37) as an attribute name, which is a SyntaxError on py37+.
- GitPython==3.1.27 requires Python 3.7 or later (requires-python:>=3.7).
```

Source: https://news.ycombinator.com/item?id=31893356

**Comment by oldkingcone** [comment]
theres another issue. 
```
ERROR: Could not find a version that satisfies the requirement GitPython==3.1.27 (from -r ./requirements.txt (line 6)) (from versions: 0.1.7, 0.2.0b1, 0.3.0b1, 0.3.0b2, 0.3.1b2, 0.3.2rc1, 0.3.2, 0.3.2.1, 0.3.3, 0.3.4, 0.3.5, 0.3.6, 0.3.7, 1.0.0, 1.0.1, 1.0.2, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.0.9.dev0, 2.0.9.dev1, 2.0.9, 2.1.0, 2.1.1, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.1.7, 2.1.8, 2.1.9, 2.1.10, 2.1.11, 2.1.12, 2.1.13, 2.1.14, 2.1.15, 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.0.4, 3.0.5, 3.0.6, 3.0.7, 3.0.8, 3.0.9, 3.1.0, 3.1.1, 3.1.2, 3.1.3, 3.1.4, 3.1.5, 3.1.6, 3.1.7, 3.1.8, 3.1.9, 3.1.10, 3.1.11, 3.1.12, 3.1.13, 3.1.14, 3.1.15, 3.1.16, 3.1.17, 3.1.18, 3.1.19, 3.1.20)
ERROR: No matching distribution found for GitPython==3.1.27 (from -r ./requirements.txt (line 6))
```

This fails on python3.6 and 3.7, however removing the version requirements for the specific packages fixes this issue.

**Comment by codaishin** [comment]
esprima requirement could be solved with `esprima==4.0.1` or - only fixing major.minor - `esprima==4.0.*`. But this project should probably always fix the whole version number (as is done currently) to prevent introduction of malicious dependencies. Maybe also provide the module hash in  `requirement.txt`, to ensure the dependencies are actually unchanged compared to what was used during development.

My local `requirement.txt` after adding all the hashes (collected today):
```
asttokens==2.0.5 --hash=sha256:0844691e88552595a6f4a4281a9f7f79b8dd45ca4ccea82e5e05b4bbdb76705c
dnspython==2.2.1 --hash=sha256:a851e51367fb93e9e1361732c1d60dab63eff98712e503ea7d92e6eccb109b4f
esprima==4.0.1 --hash=sha256:08db1a876d3c2910db9cfaeb83108193af5411fc3a3a66ebefacd390d21323ee
func_timeout==4.3.5 --hash=sha256:74cd3c428ec94f4edfba81f9b2f14904846d5ffccc27c92433b8b5939b5575dd
github3.py==3.2.0 --hash=sha256:a9016e40609c6f5cb9954dd188d08257dafd09c4da8c0e830a033fca00054b0d
GitPython==3.1.27 --hash=sha256:5b68b000463593e05ff2b261acff0ff0972df8ab1b70d3cdbd41b546c8b8fc3d
networkx==2.2 --hash=sha256:45e56f7ab6fe81652fb4bc9f44faddb0e9025f469f602df14e3b2551c2ea5c8b
protobuf==4.21.1 --hash=sha256:a4c0c6f2f95a559e59a0258d3e4b186f340cbdc5adec5ce1bc06d01972527c88
pyIsEmail==1.4.0 --hash=sha256:fbe1b3cc9bc56d244e44dd536669533fef552b46cd1f45c6dba6492303ae00c8
python_dateutil==2.8.2 --hash=sha256:961d03dc3453ebbc59dbdea9e4e11c5651520a876d0f4db161e8674aae935da9
python_magic==0.4.26 --hash=sha256:b978c4b69a20510d133a7f488910c2f07e7796f1f31703e61c241973f2bbf5fb
pytz==2018.3 --hash=sha256:07edfc3d4d2705a20a6e99d97f0c4b61c800b8232dc1c04d87e8554f130148dd
rarfile==3.0 --hash=sha256:e816409e3b36794507cbe0b678bed3e4703d7412c5f7f9201a510ed6fdc66c35
requests==2.18.4 --hash=sha256:6a1b267aa90cac58ac3a765d067950e7dbbf75b1da07e895d1f594193a40a38b
six==1.11.0 --hash=sha256:832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb
tldextract==3.1.2 --hash=sha256:f55e05f6bf4cc952a87d13594386d32ad2dd265630a8bdfc3df03bd60425c6b0
```

**Comment by Flashwalker** [comment]
yeap.
